frontend/src/components/AuthUtils/index.test.js

```diff
@@ -0,0 +1,78 @@
+import authUtils from './index.js';
+import { getCurrentUser } from './AuthUtilsConsumer.js';
+import { SIGN_IN } from '../../constants/routes.js';
+
+const mockDisplayName = 'Sam Smith';
+const mockEmail = 'samsmith@email.com';
+const mockPhotoURL = 'http://www.photo-hosters.com/qwerty';
+const mockUid = '1234567890';
+const mockAuthenticatedUser = {
+  displayName: mockDisplayName,
+  email: mockEmail,
+  photoURL: mockPhotoURL,
+  uid: mockUid
+};
+const mockUnauthenticatedUser = null;
+
+jest.mock('./AuthUtilsConsumer.js', () => ({
+  getCurrentUser: jest.fn()
+}));
+
+describe('auth utility functions when authenticated', () => {
+  beforeEach(() => {
+    getCurrentUser.mockImplementation(() => mockAuthenticatedUser);
+    expect(getCurrentUser()).toBe(mockAuthenticatedUser);
+  })
+
+  test('getUserDisplayName function', () => {
+    expect(authUtils.getUserDisplayName()).toBe(mockDisplayName);
+  });
+
+  test('getUserEmail function', () => {
+    expect(authUtils.getUserEmail()).toBe(mockEmail);
+  });
+
+  test('getUserPhotoUrl function', () => {
+    expect(authUtils.getUserPhotoUrl()).toBe(mockPhotoURL);
+  });
+
+  test('getUserUid function', () => {
+    expect(authUtils.getUserUid()).toBe(mockUid);
+  });
+});
+
+// All utility functions in this scenario should do the same thing: redirect to
+// the SIGN_IN page.
+describe('auth utility functions when not authenticated', () => {
+  beforeAll(() => {
+    delete window.location;
```

Comment:
Why is this deleting of `window.location` necessary if you are going to set it to something before each test anyway? In the case that it is something non-obvious (it was a little to me) I may recommend adding a comment to explain why you do this.

Reply:
`window.location` is set to something default in Jest tests that typically can't be modified for some reason. I believe there are other ways to get around this, such as using `Object.defineProperty` but I opted to just delete it entirely and provide my own mock version for testing. I'll add a comment to it for clarification though.